microapps/ArTabletopApp/app/src/main/java/com/arcgismaps/toolkit/artabletopapp/MainActivity.kt

```diff
@@ -28,23 +29,63 @@ import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
 import androidx.compose.material3.TopAppBar
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.res.stringResource
 import com.arcgismaps.ApiKey
 import com.arcgismaps.ArcGISEnvironment
 import com.arcgismaps.toolkit.artabletopapp.screens.MainScreen
 import com.esri.microappslib.theme.MicroAppTheme
+import com.google.ar.core.ArCoreApk
+import kotlinx.coroutines.flow.MutableStateFlow
 
 class MainActivity : ComponentActivity() {
+
+    private var userRequestedInstall: Boolean = true
+
+    // Flow to track if Google Play Services for AR is installed on the device
+    // By using `collectAsState()` in the composable, the UI will recompose when the value changes
+    private val isGooglePlayServicesArInstalled = MutableStateFlow(false)
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
         setContent {
             MicroAppTheme {
-                ArTabletopApp()
+                if (isGooglePlayServicesArInstalled.collectAsState().value) {
```

Comment:
It may not make a big difference in this specific case but still good practice to use `collectAsStateWithLifecycle`:
```suggestion
                if (isGooglePlayServicesArInstalled.collectAsStateWithLifecycle().value) {
```

Note, you would also need to update the comment above.